### ISSUE TITLE
type object got multiple values for keyword argument 'loss'

### ISSUE THREAD

**Issue by chrismaliszewski** [PyTorchLightning/pytorch-lightning#3756]
## 🐛 Bug
The error appears when TrainReport has minimize param set and loss log added at the same time with prog_bar=True

#### Code sample
```
def training_step(self, batch, batch_idx):
        loss = self(batch)
        result = pl.TrainResult(minimize=loss)
        result.log("loss", loss, prog_bar=True)

        return result
```

### Where the problem is
I followed the code and it comes to the problem with the ProgressBar callback inside progress.py line 339 -> trainer.py line 884 (return dict(**ref_model.get_progress_bar_dict(), **self.progress_bar_metrics)) which returns
```
ref_model.get_progress_bar_dict()
Out[4]: {'loss': '0.692', 'v_num': 9}
self.progress_bar_metrics
Out[5]: {'loss': 0.6924866437911987}
```

### Expected behavior
Not sure. At least the error message should be a bit clearer since a user does not create two loss logs but just one. 

### Environment
* CUDA:
	- GPU:
	- available:         False
	- version:           None
* Packages:
	- numpy:             1.19.1
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0
	- pytorch-lightning: 0.9.0
	- tqdm:              4.49.0
* System:
	- OS:                Windows
	- architecture:
		- 64bit
		- WindowsPE
	- processor:         Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
	- python:            3.8.5
	- version:           10.0.18362

**Comment by ydcjeff** [comment]
I believe TrainResult always shows the value passed to the `minimize` argument in the progress bar. So, you probably don't need to use `result.log` to show the loss in the progress bar.

**Comment by chrismaliszewski** [comment]
That's true but I think the problem is the error's message. 

As I said, a user may not realise why the error appears since they created just one log ```loss``` nor that it comes from ```prog_bar=True```. Just saying. 

The only idea that I have would be to remove showing ```minimize``` loss in progress bar by default and let users set ```loss``` log to be shown with ```prog_bar```. That would eliminate the problem. Don't know that's something others would want though.

**Comment by ydcjeff** [comment]
Thank you for your feedback.

Lightning believes almost all of users may want to show the training loss in the progress bar to be able to check/iterate the model quickly (like in the situation of overfitting the small batch) without the need to use loggers or something extra, so it is done by default and let the users use `result.log` or `self.log` for the other metrics or the various losses to show in the progress bar or in the loggers by epoch or step level.

NOTE: `self.log` is feature from the master branch.
Docs: 
- https://pytorch-lightning.readthedocs.io/en/latest/new-project.html#logging
- https://pytorch-lightning.readthedocs.io/en/latest/lightning_module.html#logging-methods

**Comment by chrismaliszewski** [comment]
Got it. I'll close the issue. It'll be for other's people reference if they have the same problem.
